src/controllers/bulk/bulkcontroller.cpp

```diff
@@ -98,16 +98,22 @@ QString BulkController::mappingExtension() {
     return BULK_MAPPING_EXTENSION;
 }
 
-void BulkController::visit(const LegacyMidiControllerMapping* mapping) {
-    Q_UNUSED(mapping);
-    // TODO(XXX): throw a hissy fit.
-    qWarning() << "ERROR: Attempting to load a LegacyMidiControllerMapping to an HidController!";
+void BulkController::setMapping(std::shared_ptr<LegacyControllerMapping> pMapping) {
+    VERIFY_OR_DEBUG_ASSERT(pMapping.use_count() == 1) {
```

Comment:
// We cannot take ownership if pMapping is referenced elsewhere

Reply:
https://en.cppreference.com/w/cpp/memory/shared_ptr/use_count

> comparison with 1. If use_count returns 1, there are no other owners. (The deprecated member function unique() is provided for this use case.) In multithreaded environment, this does not imply that the object is safe to modify because accesses to the managed object by former shared owners may not have completed, and because new shared owners may be introduced concurrently, such as by std::weak_ptr::lock. 

:upside_down_face: